### PR TITLE
community: Fix CSVLoader columns is None

### LINKED ISSUE #20699: Called strip() method with 'NoneType' as `str`
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

```python
import csv
from langchain.document_loaders.csv_loader import CSVLoader


def read_csv(path):
    csv_loader = CSVLoader(file_path=path)
    csv_data = csv_loader.load()

    return csv_data


def generate_csv_file(path):
    csv_data = [
        ['column1', 'column2', 'column3'],
        ['value1', 'value2', 'value3', 'value4', 'value5'],
        ['value6', 'value7', 'value8', 'value9']
    ]

    with open(path, 'w', newline='', encoding='utf-8') as csvfile:
        writer = csv.writer(csvfile)
        for row in csv_data:
            writer.writerow(row)


if __name__ == '__main__':
    file_path = r"example.csv"
    # 1. generate csv file
    generate_csv_file(file_path)
    # 2.read csv by CSVLoader
    data = read_csv(file_path)
```

### Error Message and Stack Trace (if applicable)

Traceback (most recent call last):
  File "C:\ProgramData\anaconda3\envs\py_310\lib\site-packages\langchain_community\document_loaders\csv_loader.py", line 68, in lazy_load
    yield from self.__read_file(csvfile)
  File "C:\ProgramData\anaconda3\envs\py_310\lib\site-packages\langchain_community\document_loaders\csv_loader.py", line 99, in __read_file
    content = "\n".join(
  File "C:\ProgramData\anaconda3\envs\py_310\lib\site-packages\langchain_community\document_loaders\csv_loader.py", line 100, in <genexpr>
    f"{k.strip()}: {v.strip() if v is not None else v}"
AttributeError: 'NoneType' object has no attribute 'strip'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "D:\code\python\llm\langchain_workstation\fix bug\bug_reproduction.py", line 31, in <module>
    data = read_csv(file_path)
  File "D:\code\python\llm\langchain_workstation\fix bug\bug_reproduction.py", line 8, in read_csv
    csv_data = csv_loader.load()
  File "C:\ProgramData\anaconda3\envs\py_310\lib\site-packages\langchain_core\document_loaders\base.py", line 29, in load
    return list(self.lazy_load())
  File "C:\ProgramData\anaconda3\envs\py_310\lib\site-packages\langchain_community\document_loaders\csv_loader.py", line 84, in lazy_load
    raise RuntimeError(f"Error loading {self.file_path}") from e
RuntimeError: Error loading example.csv

### Description

I wanted to use `langchain.document_loaders.csv_loader` to load a CSV file, but when I tried to read it through `func: read_csv` in the above **'Example Code'**, I encountered an error message as shown in the **'Error Message and Stack Trace'**.

Note: I am introducing the 'csv' library only to generate specific csv files to reproduce this bug.

### System Info

langchain==0.1.16
langchain-community==0.0.33
langchain-core==0.1.44
langchain-text-splitters==0.0.1
langsmith==0.1.49
csv==1.0

### PR DESCRIPTION
**Pull request by maang-h**
- **Bug code**: In langchain_community/document_loaders/csv_loader.py:100

- **Description**: currently, when 'CSVLoader' reads the column as None in the 'csv' file, it will report an error because the 'CSVLoader' does not verify whether the column is of str type and does not consider how to handle the corresponding 'row_data' when the column is' None 'in the csv. This pr provides a solution.

- **Issue:**  Fix #20699 

- **thinking:**

1. Refer to the processing method for 'langchain_community/document_loaders/csv_loader.py:100' when **'v'** equals'None', and apply the same method to '**k**'. (Reference`csv.DictReader` ,**'k'** will only be None when ` len(columns) < len(number_row_data)`  is established)
2. **‘k’** equals None only holds when it is the last column, and its corresponding **'v'** type is a list. Therefore, I referred to the data format in 'Document' and used ',' to concatenated the elements in the list.(But I'm not sure if you accept this form, if you have any other ideas, communicate)

### CODE REVIEW

**Comment by maang-h** [comment]
@eyurtsev @baskaryan Please help verify, thks

File: libs/community/langchain_community/document_loaders/csv_loader.py [hunk 1]
```diff
@@ -97,7 +97,9 @@ def __read_file(self, csvfile: TextIOWrapper) -> Iterator[Document]:
                     f"Source column '{self.source_column}' not found in CSV file."
                 )
             content = "\n".join(
-                f"{k.strip()}: {v.strip() if v is not None else v}"
+                f"""{k.strip() if k is not None else k}: {v.strip()
+                if v is isinstance(v, str) else ','.join(map(str.strip, v))
```
Comment (baskaryan) [inline libs/community/langchain_community/document_loaders/csv_loader.py:101]: just `if isinstance(v, str)` not `if v is isinstance(v, str)`

Comment (maang-h) [reply libs/community/langchain_community/document_loaders/csv_loader.py:101]: Oh, I will correct it.

Comment (maang-h) [reply libs/community/langchain_community/document_loaders/csv_loader.py]: @baskaryan I have corrected it, thank you for your help.

**Comment by maang-h** [comment]
@baskaryan  Since this is my first time submitting a PR, I overlooked many processes before (such as not running 'make test' locally to fully validate the code). However, this time I did a 'make test' validation locally, and although it still had a few 'FAILED', I believed it was a problem with my local environment. Therefore, I clone the project in the 'langchain ' main repository and conducted a 'make test' validation using the same environment as the code I modified, which is the same' FAILED'; Therefore, I conclude that it is a problem with my local configuration environment, not with my code. 
Please help verify again, thank you.

**Comment by maang-h** [comment]
@eyurtsev Help!!!

**Comment by maang-h** [comment]
@baskaryan After I submitted the PR, a new PR merge caused my unit test to fail, and now I have corrected it.

**Comment by maang-h** [comment]
@baskaryan um... Okay, I forgot to format the code before, but I checked them before submitting this time.  Please!!!

**Review by hwchase17 — APPROVED** [review]

**Comment by maang-h** [comment]
@baskaryan Do I need to do anything else now?

**Comment by maang-h** [comment]
@hwchase17 @baskaryan  Do I need to do anything else now?

**Pull request merged**